hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/pagination/IngresLimitHandler.java

```diff
@@ -66,7 +66,7 @@ protected boolean renderOffsetRowsKeyword() {
 	};
 
 	private static final Pattern WITH_OPTION_PATTERN =
-			Pattern.compile("\\s+with\\s+(" + String.join("|", WITH_OPTIONS) + ")\\b|\\s*(;|$)");
+			Pattern.compile("\\s+with\\s+(" + String.join("|", WITH_OPTIONS) + ")\\b|\\s*;?$");
```

Comment:
Would probably be beset to accept whitespace after the semicolon, I suppose.

Reply:
I think the sql should be trimmed before reaching here.

Reply:
Oh OK, then an alternative would be: `|\\s*;|$`, which is perhaps a little clearer .... maybe?

Reply:
`\\s*;|$` will match quoted semicolon also, perhaps you mean `\\s*;$|$`?

Reply:
Oh, right, sure, of course, I forgot what you were trying to achieve here.

Reply:
> I think the sql should be trimmed before reaching here.

Wait, are you asserting that it *does* get trimmed, or are you saying you think it *should*, but that it actually doesn't?

Reply:
> I'm not sure it actually does get trimmed or not.
> I'm asserting that it's trimmed before reaching here.

I don't understand. Those two sentences seem to contradict each other.

Reply:
https://github.com/hibernate/hibernate-orm/pull/8270/files#diff-82086e5bcedd2e6969ed098a04c054677391921bb15ad6fc4abfe4cc8784f85eL91

SQL get trimmed in the only caller `DeferredResultSetAccess`, is it appropriate? and I added a dedicated tests in `QueryTests` to guard that.

Reply:
I mean.... sure, I suppose that's fine, but I don't understand why you didn't just go with my initial suggestion to use `|\\s*;?\\s*;$` or whatever. It's simpler and more self-contained, and so much less-likely to be break in future.

I don't see why this fix should be impacting code in `DeferredResultSetAccess`, which is rather a long way away.

Reply:
Amended as you suggested.